packages/core/src/core.ts

```diff
@@ -97,7 +97,11 @@ export function exportVariable(name: string, val: any): void {
  * @param secret value of the secret
  */
 export function setSecret(secret: string): void {
-  issueCommand('add-mask', {}, secret)
+  for (const part of secret.split(/[\r\n]/)) {
```

Comment:
minor optimization would be to add a `+` outside the square brackets.  That should reduce the number of empty strings the following conditional has to evaluate (particularly on Windows).

Reply:
Great shout, sorry I missed it in your previous comment!